### PR TITLE
perf(stats): three opt-in micro-optimizations (simdutf8 output, t-digest quantiles, mode-cardinality cap)

### PR DESCRIPTION
**Pull request by jqnatividad**
## Summary

Three independently-shippable performance opportunities for `qsv stats`,
all opt-in or behavior-preserving on default flags. Each landed as a
discrete commit so they can be reverted or cherry-picked individually.

- **`9e937e50e` — `simdutf8` + `Cow` in output**: replaces 7
  `String::from_utf8_lossy` sites in `to_record` and `format_antimodes`
  with a `bytes_to_cow_str` helper that uses SIMD-validated UTF-8 and
  returns a borrowed `&str` on the happy path. Byte-identical for valid
  UTF-8 (the entire test universe); only difference is faster validation.
- **`e99fe9b5b` — `--quantile-method approx` (t-digest)**: opt-in
  approximate-quantile engine for `--median`, `--quartiles`, and
  `--percentiles`, backed by the Apache DataSketches t-digest port
  (`datasketches` crate). Trades exact for O(K) memory (~200 centroids)
  and O(1) quantile reads with ~1% rank error. Default `exact` produces
  byte-identical output to today, asserted by a regression test.
  Restrictions enforced at parse time: `--weight` is rejected (upstream
  `TDigestMut` has no public weighted-update); `--mad` warns and is
  disabled (algorithmically incompatible with t-digest).
- **`b20a7a407` — `--mode-cardinality-cap N`**: opt-in upper bound on
  mode-tracking memory. When `N > 0`, qsv drops the mode tracker for any
  column whose `Unsorted::len()` (unweighted, total samples) or
  `HashMap::len()` (weighted, unique keys) exceeds `N`, and emits
  sentinels (`*HIGH_CARDINALITY` / `>=N`) instead of crashing or
  truncating silently. Default `0` = unbounded (today's behavior).

Both new flags invalidate the stats cache via `StatsArgs` JSON sidecar
when changed, so cached output never silently mismatches the requested
algorithm.

`5e6d92e39` and `fc91fcba5` address LOW review findings on the t-digest
and cap commits respectively (review jobs 2002 and 2003) — comment
cleanups, the `--jobs 1` determinism note, dropped unused Serde impls,
single-invocation test capture, plus added coverage for the
weighted-cap branch, the multi-chunk post-merge re-check, and the
`--cardinality`-only sentinel-field count.

Help docs auto-regenerated via `qsv --generate-help-md`.

Diff: 972 insertions / 195 deletions across 6 files; the bulk of the
deletions are formatter realignments triggered by widening the field-name
column in `Args` / `StatsArgs` / `WhichStats`.

## Test plan

- [x] `cargo build --locked --bin qsv -F all_features`
- [x] `cargo test --locked stats -F all_features` — 740 passed, 0 failed
      (728 baseline + 5 t-digest + 6 mode-cap + 1 empty-numeric edge case)
- [x] `cargo clippy --locked --bin qsv -F all_features -- -D warnings` — clean
- [x] Smoke test approx mode against exact mode on uniform 1..=10_000:
      q1/q2/q3 within 2% envelope
- [x] Smoke test cap=3 on a 10-row file: `*HIGH_CARDINALITY` and `>=3`
      emitted; cap=100 on the same file: normal output
- [x] Smoke test parse-time rejection: `--weight` + `--quantile-method approx`
      errors out with a clear message; invalid `--quantile-method nonsense`
      errors out
- [ ] Reviewer: please run `scripts/benchmarks.sh` to quantify the perf
      delta on the NYC 311 1M-row dataset (didn't run locally because the
      script requires the dataset to be present)
- [ ] Reviewer: please confirm `--quantile-method approx` output reads
      sensibly under multi-thread (`--jobs 4`) — the test suite pins
      `--jobs 1` for deterministic snapshots; `--jobs >= 2` is documented
      as approximate-only

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/cmd/stats.rs [hunk 1]
```diff
@@ -3315,15 +3472,31 @@ impl Stats {
...
+        // sentinels. cap == 0 (default) means unbounded — preserves today's behavior.
+        let cap = self.which.mode_cardinality_cap;
         if let Some(ref mut wm) = self.weighted_modes {
-            // Weighted modes: accumulate weights per value
+            // Weighted modes: HashMap::len() == unique-values seen (true cardinality).
+            if cap > 0 && wm.len() as u64 > cap {
+                self.weighted_modes = None;
+                self.modes_dropped = true;
+                return;
+            }
```
[the remark this answers is not in the document]
Comment (jqnatividad) [reply src/cmd/stats.rs]: Applied. Tightened the cap check to `>= cap` BEFORE insert (was `> cap`, which allowed peak = cap+1). For the weighted path the check now sits inside the new-key branch only, so existing-key updates (which don't grow the map) skip the cap and the get_mut fast-path is preserved. Peak retained is now exactly `cap`.

File: src/cmd/stats.rs [hunk 2]
```diff
@@ -2650,19 +2729,29 @@ fn resolve_sniff_whitelist(input_path: &std::path::Path) -> CliResult<String> {
...
+    median:               bool,
+    mad:                  bool,
+    quartiles:            bool,
+    mode:                 bool,
+    typesonly:            bool,
+    percentiles:          bool,
+    use_weights:          bool,
+    /// When true, use the Apache DataSketches t-digest engine for median, quartiles, and
+    /// custom percentiles instead of the exact (sort-based) `Unsorted<f64>` engine. Mutually
+    /// exclusive with `mad` and `use_weights`; validation in `Args::run()` rejects the bad
```
[the remark this answers is not in the document]
Comment (jqnatividad) [reply src/cmd/stats.rs]: Applied. Replaced the `Args::run()` reference with "module-level `run(argv: ...)` function" — there is no `Args::run()` method; validation lives in the free `run` function in this module.

File: tests/test_stats.rs [hunk 3]
```diff
@@ -6052,3 +6052,487 @@ fn stats_jsonl_tsv_delimiter() {
...
+    cmd.arg("--quartiles")
+        .arg("--quantile-method")
+        .arg("approx")
+        .arg("--jobs")
+        .arg("1")
+        .arg("data.csv");
+
+    wrk.assert_success(&mut cmd);
+
+    let got: Vec<Vec<String>> = wrk.read_stdout(&mut cmd);
```
[the remark this answers is not in the document]
Comment (jqnatividad) [reply tests/test_stats.rs]: Applied. Replaced `wrk.assert_success(&mut cmd)` + `wrk.read_stdout(&mut cmd)` with a single `cmd.output()` capture; success is asserted on that output, and headers/value row are parsed from stdout bytes. Same fix as `stats_quantile_method_approx_disables_mad`.

File: tests/test_stats.rs [hunk 4]
```diff
@@ -6052,3 +6052,487 @@ fn stats_jsonl_tsv_delimiter() {
...
+    cmd.arg("--percentiles")
+        .arg("--quantile-method")
+        .arg("approx")
+        .arg("--jobs")
+        .arg("1")
+        .arg("data.csv");
+
+    wrk.assert_success(&mut cmd);
+
+    let got: Vec<Vec<String>> = wrk.read_stdout(&mut cmd);
```
[the remark this answers is not in the document]
Comment (jqnatividad) [reply tests/test_stats.rs]: Applied. Single `cmd.output()` capture with status assertion on the captured output; stdout parsed from the bytes. Matches the pattern used in the other approx-mode tests.
